doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py

```diff
@@ -3,16 +3,20 @@
 import numpy
 
 import Orange.data
-from Orange.widgets import widget, gui
+from Orange.widgets.widget import OWWidget, Input, Output
+from Orange.widgets import gui
 
-class OWDataSamplerA(widget.OWWidget):
+class OWDataSamplerA(OWWidget):
     name = "Data Sampler"
     description = "Randomly selects a subset of instances from the data set"
     icon = "icons/DataSamplerA.svg"
     priority = 10
 
-    inputs = [("Data", Orange.data.Table, "set_data")]
-    outputs = [("Sampled Data", Orange.data.Table)]
+    class Inputs:
+        data = Input("Data", Orange.data.Table)
+
+    class Outputs:
+        sample = Output("Sampled Data", Orange.data.Table)
```

Comment:
Correction, Inputs/Outpus need not extend anything. This would be OWWidget metaclass whose namespace dict reinterprets what it really means to be Inputs. Indeed, magic. (But the interface must be there for a reason.)